pkg/apis/v1alpha1/edge-placement.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// EdgePlacement binds a collection of Namespaces and non-namespaced objects
+// to a collection of Locations.
+// An EdgePlacement object appears in the center and directs all the syncable
+// objects in the selected Namespaces and all the selected non-namespaced objects
+// to propagate to _all_ of the selected Locations
+// (one SyncTarget in each such Location).
+// This is not entirely unrelated to a TMC Placement, which directs the selected
+// Namespaces to propagate to _one_ of the selected Locations.
+//
+// Overlap between EdgePlacements is allowed:
+// two different EdgePlacement objects may select intersecting Location sets
+// and/or intersecting Namespace sets.
+// This is not problematic because:
+//   - propagation _into_ a destination is additive;
+//   - propagation _from_ a source is additive;
+//   - two directives to propagate the same object to the same destination are
+//     simply redundant (they, by design, can not conflict).
+type EdgePlacement struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// `spec` describes a desired binding between Namespaces and Locations.
```

Comment:
Namespaces _and non-namespaced objects_?

Reply:
Fixed in latest revision.